include/pybind11/detail/common.h

```diff
@@ -1178,5 +1237,24 @@ constexpr inline bool silence_msvc_c4127(bool cond) { return cond; }
 #    define PYBIND11_DETAILED_ERROR_MESSAGES
 #endif
 
+#if (defined(__GNUC__) && __GNUC__ == 4 && !defined(__clang__))
+
+// From https://gcc.gnu.org/bugzilla/show_bug.cgi?id=57350#c11
+inline void *align(std::size_t alignment, std::size_t size, void *&ptr, std::size_t &space) {
+    std::uintptr_t pn = reinterpret_cast<std::uintptr_t>(ptr);
+    std::uintptr_t aligned = (pn + alignment - 1) & -alignment;
+    std::size_t padding = aligned - pn;
+    if (space < size + padding)
+        return nullptr;
+    space -= padding;
+    return ptr = reinterpret_cast<void *>(aligned);
```

Comment:
```suggestion
    return ptr = static_cast<void *>(aligned);
```

Reply:
This isn't support on centos7 for some reason.